bot/cogs/watchchannels/talentpool.py

```diff
@@ -220,6 +206,36 @@ async def edit_reason_command(self, ctx: Context, nomination_id: int, *, reason:
 
         await ctx.send(f":white_check_mark: Updated the {field} of the nomination!")
 
+    @Cog.listener()
+    async def on_member_ban(self, guild: Guild, user: Union[User, Member]) -> None:
+        """Remove `user` from the talent pool after they are banned."""
+        await self.unwatch(user.id, "User was banned.")
+
+    async def unwatch(self, user_id: int, reason: str) -> bool:
+        """End the active nomination of a user with the given reason and return True on success."""
+        active_nomination = await self.bot.api_client.get(
+            self.api_endpoint,
+            params=ChainMap(
+                self.api_default_params,
+                {"user__id": str(user_id)}
+            )
+        )
+
+        if not active_nomination:
+            log.debug(f"No active nominate exists for {user_id=}")
+            return False
+
+        log.info(f"Ending nomination: {user_id=} {reason=}")
+
+        [nomination] = active_nomination
```

Comment:
This line triggers me a little bit. I don't really see why you have to do this. `active_nomination['id']` in the patch call wouldn't really be that bad, and the `[x] = y` syntax just looks deeply wrong to me no matter how many times I come across it.